File: resource/interface/client/objective.lua
local isOpen = false

---@param title string
---@param content string
function prp.showObjective(title, content)
    if not title or not content then return end
    
    isOpen = true
    SendNUIMessage({
        action = 'showObjective',
        data = {
            title = title,
            content = content
        }
    })
end

function prp.hideObjective()
    isOpen = false
    SendNUIMessage({ action = 'hideObjective' })   
end

function prp.isObjectiveActive()
    return isOpen
end

File: web/src/features/dev/debug/objective.ts
import type { ObjectiveProps } from "../../../typings/objective";
import { debugData } from "../../../utils/debugData";

export const debugObjective = () => {
    debugData<ObjectiveProps>([
        {
            action: 'showObjective',
            data: {
                title: 'Wait for client',
                content: 'Wait for client\'s offer. You\'ll recieve an offer soon.'
            }
        }
    ])
}